src/nlp/Preprocessing.py

```diff
@@ -0,0 +1,477 @@
+        unknowns = list(set1.union(set2))
+
+    corrections={}
+    for word in unknowns:
+      if word.lower() in abbrDict.keys():
+        if len(abbrDict[word.lower()]) > 0:
+          corrections[word] = abbrDict[word.lower()]
+      else:
+        # Here we are addressing the fact that the abbreviation database will never be complete
+        # Given an abbreviation that is not part of the abbreviation database, we are looking for a
+        # a subset of abbreviations the abbreviation database that are close enough (and consider
+        # them as possible candidates
+        from difflib import SequenceMatcher
+        corrections[word] = []
+        abbreviationDS = list(abbrDict)
+        for index,abbr in enumerate(abbreviationDS):
+          val=0
+          newVal = SequenceMatcher(None, word, abbr).ratio()
+          if newVal>=0.75 and newVal>val:
+            corrections[word] = abbrDict[abbr]
+            val = newVal
+      if not corrections[word]:
+        corrections.pop(word)
+
+    combinations = list(itertools.product(*list(corrections.values())))
+    options = []
+    for comb in combinations:
+      corrected = text
+      for index,key in enumerate(corrections.keys()):
+        corrected = re.sub(r"\b%s\b" % str(key) , comb[index], corrected)
+      options.append(corrected)
+
+    if not options:
+      return text
+    else:
+      bestOpt = self.findOptimalOption(options)
+      return bestOpt
+
+  def findOptimalOption(self,options):
+    """
+      Method to handle abbreviation with multiple meanings
+      @ In, options, list, list of sentence options
+      @ Out, optimalOpt, string, option from the provided options list that fits more the
+                                 possible
+    """
+    nOpt = len(options)
+    combScore = np.zeros(nOpt)
+    for index,opt in enumerate(options):
+      listOpt = opt.split()
+      for i,word in enumerate(listOpt):
+        for j in range(i+1,len(listOpt)):
+          combScore[index] = combScore[index] + wordsSimilarity(word,listOpt[j])
+    optIndex = np.argmax(combScore)
+    optimalOpt = options[optIndex]
+    return optimalOpt
+
+
+class AbbrExpander(object):
+  """
+    Class to expand abbreviations
+  """
+
+  def __init__(self, abbreviationsFilename, checkerType='autocorrect', abbrType='mixed'):
+    """
+      Abbrviation expander constructor
+      @ In, abbreviationsFilename, string, filename of abbreviations data
+      @ Out, None
+    """
+    self.abbrType = abbrType
+    self.checkerType = checkerType
+
+    self.abbrList = pd.read_excel(abbreviationsFilename)
+    self.preprocessorList = ['hyphenated_words',
+                             'whitespace',
+                             'numerize']
+    self.preprocess = Preprocessing(self.preprocessorList, {})
+    self.checker = SpellChecker(checker=self.checkerType)
+    self.abbrDict = self.checker.generateAbbrDict(self.abbrList)
+
+
+  def abbrProcess(self, text, splitToList='False'):
```

Comment:
should this be:
`def abbrProcess(self, text, splitToList=False):`

because:
```
>>> not 'False'
False
```
so the line `if not splitToList:` will not work like expected.

Reply:
I pushed a fix.